src/python/pants/backend/python/target_types.py

```diff
@@ -1349,3 +1350,77 @@ class PythonDistribution(Target):
         See {doc_url('python-distributions')}.
         """
     )
+
+
+class SetuptoolsSCMDummySourceField(OptionalSingleSourceField):
+    """A dummy SourceField for participation in the codegen machinery."""
+
+    alias = "_dummy_source"  # Leading underscore omits the field from help.
+    help = "A version generated by setuptools_scm"
+
+
+class SetuptoolsSCMTagRegexField(StringField):
+    default = None
+    alias = "tag_regex"
+    help = softwrap(
```

Comment:
Might be worth mentioning what happens if left undefined.

Reply:
I'll just copy the default from setuptools_scm, that will be better than me trying to describe a regex in words...